src/jvm/main/org/jetbrains/kotlinx/lincheck/trace/TraceCompressor.kt

```diff
@@ -190,6 +193,28 @@ private fun SingleThreadedTable<TraceNode>.compressThreadStart() = compressNodes
     newNode
 }
 
+internal fun SingleThreadedTable<TraceNode>.collapseLibraries() = compressNodes { node -> 
+    // if should not be hidden
+    if (node !is CallNode || !shouldBeHidden(node)) return@compressNodes node
+    
+    // if cannot be hidden (due to switch point)
+    if (node.containsDescendant { it is EventNode && it.tracePoint is SwitchEventTracePoint }) 
+        return@compressNodes node
+    
+    val newNode = node.copy()
+    findNonLibrarySubTrees(node).forEach {  newNode.addChild(it) }
+    return@compressNodes newNode
+}
+
+/**
+ * Finds descendants that should not be hidden.
+ * But not descendants of descendants, aka the root of the subtree that should not be hidden.
+ */
+private fun findNonLibrarySubTrees(node: TraceNode): List<TraceNode> {
```

Comment:
Please rename. It is not specific to stdlibs, in future we might want to hide other stuff as well.